test/c/test_circuit.c

```diff
@@ -22,27 +22,52 @@
 /**
  * Test the zero constructor.
  */
-int test_empty() {
+int test_empty(void) {
     QkCircuit *qc = qk_circuit_new(0, 0);
     uint32_t num_qubits = qk_circuit_num_qubits(qc);
     uint32_t num_clbits = qk_circuit_num_clbits(qc);
     size_t num_instructions = qk_circuit_num_instructions(qc);
     qk_circuit_free(qc);
 
-    return (num_qubits != 0 || num_clbits != 0 || num_instructions != 0) ? EqualityError : Ok;
+    if (num_qubits != 0) {
+        printf("The number of qubits %d is not 0", num_qubits);
+        return EqualityError;
+    }
+    if (num_clbits != 0) {
+        printf("The number of clbits %d is not 0", num_clbits);
+        return EqualityError;
+    }
+    if (num_instructions != 0) {
+        printf("The number of instructions %lu is not 0", num_instructions);
+        return EqualityError;
+    }
+    return Ok;
 }
 
-int test_no_gate_1000_bits() {
+int test_no_gate_1000_bits(void) {
     QkCircuit *qc = qk_circuit_new(1000, 1000);
     uint32_t num_qubits = qk_circuit_num_qubits(qc);
     uint32_t num_clbits = qk_circuit_num_clbits(qc);
     size_t num_instructions = qk_circuit_num_instructions(qc);
     qk_circuit_free(qc);
 
-    return (num_qubits != 1000 || num_clbits != 1000 || num_instructions != 0) ? EqualityError : Ok;
+    if (num_qubits != 1000) {
+        printf("The number of qubits %d is not 1000", num_qubits);
```

Comment:
It would be nice to standardize this in the future. I'm wondering if this is printed somehow in order with CTest's output or if it is shown at another point and the origin is hard to pin down.